.github/workflows/pull-request.yml

```diff
@@ -182,223 +185,13 @@ jobs:
 
   Check-AutoDocs:
     name: Generate and Validate Documentation
-    runs-on: ubuntu-latest
     needs: [Code-Quality-Checks]
-    steps:
-      - name: Checkout the Repository
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - name: Install pnpm
-        uses: pnpm/action-setup@v4
-        with:
-          run_install: false
-
-      - name: Set up Node.js
-        uses: actions/setup-node@v4
-        with:
-          node-version: '24.x'
-          cache: 'pnpm'
-
-      - name: Prepare dependency store
-        run: pnpm fetch
-
-      - name: Install Dependencies (frozen)
-        run: pnpm install --frozen-lockfile --prefer-offline
-
-      - name: Run documentation generation
-        run: pnpm run generate-docs
-
-      - name: Check for uncommitted doc changes
-        run: |
-          if [ -n "$(git status --porcelain)" ]; then
-            echo "::error::Documentation files are outdated or missing."
-            echo "Please run 'pnpm run generate-docs' locally and commit the updated files."
-            echo ""
-            echo "Changed files:"
-            git status --porcelain
-            exit 1
-          else
-            echo "Documentation is up to date."
-          fi
-
-      - name: Check for unused dependencies
-        run: pnpm knip --config knip.deps.json --include dependencies
-
+    uses: PalisadoesFoundation/.github/.github/workflows/typescript-autodocs.yml@main
   Check-Sensitive-Files:
-    if: ${{ github.actor != 'dependabot[bot]' }}
-    name: Checks if sensitive files have been changed without authorization
```

Comment:
The sensitive files check will need to be excluded.
1. It varies slightly between repos
2. This issue has been created to make it more customizable. Hopefully we'll be able to place the config files in the central repository too.
    1. https://github.com/PalisadoesFoundation/talawa-admin/issues/5056